pkg/agent/controller/ipseccertificate/ipsec_certificate_controller.go

```diff
@@ -74,6 +74,8 @@ type Controller struct {
 	rotateCertificate  func() (*certificateKeyPair, error)
 	certificateKeyPair *certificateKeyPair
 
+	now func() time.Time
```

Comment:
IMO, we should use a `clock.Clock` member field here. It's cleaner than just injecting the `Now` function, and it lets us use different time functions in the future if we need to.

Reply:
Thanks for the insights!